xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -2557,61 +2558,83 @@ static std::optional<GlobalDeviceId> DeviceConstraint(
   return std::nullopt;
 }
 
-absl::Status IrEmitterUnnested::EmitCopyStartThunk(
-    const HloCopyStartInstruction* instr) {
+absl::Status
+IrEmitterUnnested::EmitCopyStartThunk(const HloCopyStartInstruction *instr) {
   // copy-start has a tuple shape: {host, device, context},
   // or {device, host, context}.
   // Only the destination shape is needed to get the output buffer.
   TF_ASSIGN_OR_RETURN(BufferAllocation::Slice dst_buffer,
                       GetAllocationSliceForHlo(instr,
                                                /*ShapeIndex=*/{0}));
 
-  const HloInstruction* src = instr->operand(0);
-  const Shape& input_shape = src->shape();
+  const HloInstruction *src = instr->operand(0);
+  const Shape &input_shape = src->shape();
   TF_ASSIGN_OR_RETURN(BufferAllocation::Slice src_buffer,
                       GetAllocationSliceForHlo(src, {}));
   Shape shape = instr->shape();
   CHECK(shape.IsTuple());
+  enum copy_direction { H2D = 0, D2H = 1, D2D = 2 };
+  copy_direction dir = D2D;
```

Comment:
Seems H2D enum value isn't actually used anywhere? Do we even need to declare a separate enum?

Reply:
H2D enum is used. D2D is not and removed.